scripts/translate_pdf.py

```
@@ -0,0 +1,725 @@
+                            
+                            # Add to translations dict and cache
+                            for text, translation in zip(batch_items, batch_results):
+                                translations[text] = translation
+                                self.cache[text] = translation
+                            
+                            self.stats['api_calls'] += 1
+                            # Periodically save cache to prevent data loss on crash
+                            if self.stats['api_calls'] % 5 == 0:
+                                self._save_cache()
+                                
+                        except Exception as exc:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)